### PR TITLE
Codegen for space view python blueprint classes

### LINKED ISSUE #6081: Codegen for SpaceView blueprint types + properties
Space view properties are fairly simple to handle: They are archetypes!

However, adding properties to the blueprint api takes a bunch of manual steps right now as illustrated in
* #6068

We should a piece of codegen that generates the classes for each space view. We can do this with another fbs table that specifies:
* space view name / type
* list of property archetypes
Furthermore, each property archetype needs to be able to define a list of types from which it can be built directly (in the Background3D examples  that would be `Rgba32Like | Background3DKind`).

Only python needs to be covered for now.

Once this is done we should wire up all existing space view property types (which will be trivial then!)

### PR DESCRIPTION
**Pull request by Wumpf**
### What

* Fixes #6081

Extends our codegen to accept a fourth object type, space views (in addition to datatypes/components/archetypes). Meaning we can do special codegen for space views now, making it easy to set them up consistently and define what properties a space view can be constructed from. Building up on the previous PR that exposed background3d color, this generates almost identical code now.

Doesn't add yet any new view properties - this is going to happen in a follow-up PR since it requires getting a few more things serializable and subsequently also testing.

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
  * Using examples from latest `main` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6100?manifest_url=https://app.rerun.io/version/main/examples_manifest.json)
  * Using full set of examples from `nightly` build: [rerun.io/viewer](https://rerun.io/viewer/pr/6100?manifest_url=https://app.rerun.io/version/nightly/examples_manifest.json)
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG
* [x] If applicable, add a new check to the [release checklist](https://github.com/rerun-io/rerun/blob/main/tests/python/release_checklist)!

- [PR Build Summary](https://build.rerun.io/pr/6100)
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

To run all checks from `main`, comment on the PR with `@rerun-bot full-check`.

### CODE REVIEW

File: rerun_py/tests/test_types/space_views/.gitattributes [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# DO NOT EDIT! This file is generated by crates/re_types_builder/src/lib.rs
+
+.gitattributes linguist-generated=true
```
Comment (Wumpf) [inline rerun_py/tests/test_types/space_views/.gitattributes:3]: there's no tests in here, this is silly

File: rerun_py/tests/unit/test_scalar_axis.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from __future__ import annotations
```
Comment (Wumpf) [inline rerun_py/tests/unit/test_scalar_axis.py:1]: oops. this is not supposed to be here yet

File: rerun_py/tests/test_types/space_views/.gitattributes (hunk 1 above)
Comment (Wumpf) [reply rerun_py/tests/test_types/space_views/.gitattributes:3]: not the only one though. not bothering with this right now

File: crates/re_types_builder/src/codegen/python/space_views.rs [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+pub fn code_for_space_view(reporter: &Reporter, objects: &Objects, obj: &Object) -> String {
+    assert!(obj.is_struct());
+
+    let mut code = String::new();
+
+    code.push_indented(
+        0,
+        "
+from .. import archetypes as blueprint_archetypes
```
Comment (jleibs) [inline crates/re_types_builder/src/codegen/python/space_views.rs:17]: this assumes space views are always in the blueprint scope. Probably a safe assumption but probably worth adding the assert somewhere.

File: crates/re_types_builder/src/objects.rs [hunk 4]
```diff
@@ -68,28 +68,34 @@ impl Objects {
             objects: resolved_enums.into_iter().chain(resolved_objs).collect(),
         };
 
-        // Validate fields types: Archetype consist of components, everything else consists of datatypes.
+        // Validate fields types: Archetype consist of components, Space Views consist of archetypes, everything else consists of datatypes.
```
Comment (jleibs) [inline crates/re_types_builder/src/objects.rs:71]: I can't help but wonder if "SuperArchetypes" (structured collections of archetype) are a generic things of which SpaceViews just happen to be the first example.

File: crates/re_types_builder/src/objects.rs [hunk 5]
```diff
@@ -68,28 +68,34 @@ impl Objects {
...
+                                reporter.error(virtpath, &obj.fqname, format!("Points to an instance of {field_type_fqname:?}) is part of an space view but is not an archetype. Only archetypes are allowed as fields of a space view's properties."));
+                            }
+                        }
                     }
                 } else {
                     // Note that we *do* allow primitive fields on components for the moment. Not doing so creates a lot of bloat.
-                    assert!(obj.kind != ObjectKind::Archetype,
-                        "{virtpath}: Field {:?} is a primitive field of type {:?}. Only Components are allowed on Archetypes.",
-                        field.fqname, field.typ);
+                    if obj.kind == ObjectKind::Archetype && obj.kind == ObjectKind::SpaceView {
```
Comment (jleibs) [inline crates/re_types_builder/src/objects.rs:96]:
```suggestion
                    if obj.kind == ObjectKind::Archetype || obj.kind == ObjectKind::SpaceView {
```
this looks like a logical error.

**Review by jleibs — APPROVED** [review]

File: crates/re_types_builder/src/codegen/python/space_views.rs (hunk 3 above)
Comment (Wumpf) [reply crates/re_types_builder/src/codegen/python/space_views.rs]: somewhat circular: we identify something bein `ObjectKind::View` if `pkg_name.starts_with("rerun.blueprint.views")` is true

**Pull request merged**